pkg/roachpb/api.proto

```diff
@@ -401,6 +405,22 @@ message ScanRequest {
   // keys returned by the request, not a single range lock over the entire span
   // scanned by the request.
   kv.kvserver.concurrency.lock.Strength key_locking = 5;
+
+  // This "any type" field is a serialized TableReaderSpec that can be used
+  // to initialize a columnar scan for MVCC. We use this generic type to resolve
+  // some painful circular dependency issues.
+  google.protobuf.Any scan_spec = 6;
```

Comment:
I suspect we'd want to do something more principled here if this is productionized.